### PR TITLE
[Merged by Bors] - feat(Topology/Bases): add lemma Embedding.separableSpace from Embedding.secondCountableTopology

### PR DESCRIPTION
**Pull request by luigi-massacci**
We prove that if a space embeds in a second-countable space, then it is separable, by weakening the same result for second-countability. Light generalization of the statement that if a metric space embeds into a separable metric space then it is itself separable.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Bases.lean [hunk 1]
```diff
@@ -1034,3 +1034,9 @@ protected theorem Embedding.secondCountableTopology [SecondCountableTopology β]
     SecondCountableTopology α :=
   hf.1.secondCountableTopology
 #align embedding.second_countable_topology Embedding.secondCountableTopology
+
+protected theorem Embedding.separableSpace {α : Type u_1} {β : Type u_2} [TopologicalSpace α]
```
Comment (Ruben-VandeVelde) [inline Mathlib/Topology/Bases.lean:1038]: You can reuse the alpha and beta from the variables

File: Mathlib/Topology/Bases.lean [hunk 2]
```diff
@@ -1034,3 +1034,9 @@ protected theorem Embedding.secondCountableTopology [SecondCountableTopology β]
...
+    [TopologicalSpace β] [SecondCountableTopology β] {f : α → β} (hf : Embedding f) :
+    TopologicalSpace.SeparableSpace α := by
+  have := Embedding.secondCountableTopology hf
+  exact TopologicalSpace.SecondCountableTopology.to_separableSpace
```
Comment (Ruben-VandeVelde) [inline Mathlib/Topology/Bases.lean:1042]: Can the proof be just `hf.secondCountableTopology.to_separableSpace`?

**Review by Ruben-VandeVelde — CHANGES_REQUESTED** [review]

File: Mathlib/Topology/Bases.lean (hunk 1 above)
Comment (luigi-massacci) [reply Mathlib/Topology/Bases.lean:1038]: Indeed.

File: Mathlib/Topology/Bases.lean (hunk 2 above)
Comment (luigi-massacci) [reply Mathlib/Topology/Bases.lean:1042]: Complains that the environment does not contain 'SecondCountableTopology.to_separableSpace'

Comment (Ruben-VandeVelde) [reply Mathlib/Topology/Bases.lean]: Oh, that's silly, but I now realize it wouldn't work anyway.

**Review by Ruben-VandeVelde — APPROVED** [review]
Thanks!

maintainer merge

**Pull request closed without merging**